frontend/src/pages/Root.js

```diff
@@ -0,0 +1,38 @@
+import { useEffect } from 'react';
+import { Outlet, useLoaderData, useSubmit } from 'react-router-dom';
+
+import MainNavigation from '../components/Header/MainNavigation';
+import { getTokenDuration } from '../util/auth';
+
+function RootLayout() {
+  const token = useLoaderData();
+  const submit = useSubmit();
+  useEffect(() => {
+    if (!token) {
+      return;
+    }
+
+    if (token === 'EXPIRED') {
+      submit(null, { action: '/logout', method: 'post' });
+      return;
+    }
+
+    const tokenDuration = getTokenDuration();
+    console.log(tokenDuration);
+
+    setTimeout(() => {
+      submit(null, { action: '/logout', method: 'post' });
+    }, tokenDuration);
```

Comment:
clear the timeout when the component is unmounted to avoid potential memory leaks.